File: mario.c
#include <stdio.h>
#include <cs50.h>

int main(void)
{
    int n;

    do
    {
        n = get_int("Height: ");
    }
    while (n < 0 || n > 23);
    // "bricks" denotes the number of blocks to be printed on each row
    for (int bricks = 0; bricks <= n - 1; bricks++)
    {
        // the number of spaces printed is the height minus the number of bricks
        for (int j = 0; j < n - bricks - 1; j++)
        {
            printf(" ");
        }
        // we want two bricks on the zeroth row, so we add 2 to do so
        for (int k = 0; k < bricks + 2; k++)
        {
            printf("#");
        }
        printf("\n");
    }
}

File: mario.py
from cs50 import get_int


def main():
    # Keep prompting user for height of pyramid until they give valid number
    while True:
        height = get_int("Height: ")
        if height >= 0 and height <= 23:
            break
    # In each row, print a specfic number of bricks and spaces that is dependent on the row number
    for i in range(height):
        print(" " * (height - i - 1), end="")
        print("#" * (i + 2))


if __name__ == "__main__":
    main()
